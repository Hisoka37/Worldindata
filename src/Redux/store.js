import { configureStore } from '@reduxjs/toolkit';
import countriesReducer from './countries/countriesSlice';

export const store = configureStore({
  reducer: {
    country: countriesReducer,
  },
});
export default store;
