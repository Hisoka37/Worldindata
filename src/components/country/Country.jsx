import { useEffect } from 'react';
import './country.css';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import {
  displayAllcountries,
  searchByRegion,
} from '../../Redux/countries/countriesAction';

const Country = () => {
  const {
    countriesData, success, loading, error, region, searchTerm,
  } = useSelector((store) => store.country);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(displayAllcountries());
    if (region) {
      dispatch(searchByRegion(region));
    }
    if (error) {
      console.log(error);
    }
  }, [dispatch, error, success, region]);

  const data = countriesData.filter((item) => item.name.common.toLowerCase().includes(searchTerm.toLowerCase()));

  return (
    <section className="country-container">
      {loading ? (
        <h1>Loading .....</h1>
      ) : (
        data.map((item) => (
          <Link className="country-card" key={item.id} to={`/${item.cioc}`}>
            <img
              src={item.flags.png}
              alt={item.flags.alt}
              className="country-image"
            />
            <div className="country-content">
              <div className="countryheader">
                <h3>{item.name.common}</h3>
                <i className="fa fa-arrow-circle-o-right" />
              </div>
              <p>
                Population:
                {' '}
                <span>{item.population}</span>
              </p>
            </div>
          </Link>
        ))
      )}
    </section>
  );
};

export default Country;
