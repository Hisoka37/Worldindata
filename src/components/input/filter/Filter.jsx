import { useEffect, useState } from "react";
import "./filter.css";
import { useDispatch } from "react-redux";
import { reset, setRegion } from "../../../Redux/countries/countriesSlice";

const Filter = () => {
  const regions = ["Africa", "Americas", "Asia", "Europe", "Oceania"];
  const [filter, setFilter] = useState("");
  const [displayDropDown, setDisplayDropdown] = useState(false);

  const dispatch = useDispatch();
  const handleDrodown = () => {
    setDisplayDropdown(!displayDropDown);
  };

  useEffect (() => {
    if (filter !== "") {
      dispatch(setRegion(filter.toLowerCase()));
    }
  }, [dispatch, filter])


  return (
    <section className="filter-container">
      <div className="filter">
        <input
          type="text"
          readOnly
          placeholder="Filter by Region"
          value={filter}
          className="filter-input"
        />

        <i className="fa fa-arrow-down" onClick={handleDrodown}></i>
      </div>
      {displayDropDown ? (
        <div className="dropdown">
          {regions.map((item, index) => {
            return (
              <div
                key={index}
                className="dropdown-item"
                onClick={() => {
                  setFilter(item);
                  handleDrodown();
                }}
              >
                {item}
              </div>
            );
          })}
        </div>
      ) : null}
    </section>
  );
};

export default Filter;
