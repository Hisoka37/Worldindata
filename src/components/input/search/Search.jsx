import { useDispatch, useSelector } from 'react-redux';
import './search.css';
import { setSearchTerm } from '../../../Redux/countries/countriesSlice';

const Search = () => {
  const { searchTerm } = useSelector((state) => state.country);
  const dispatch = useDispatch();
  const handleInputValue = (e) => {
    dispatch(setSearchTerm(e.target.value.toLowerCase()));
  };
  return (
    <section className="search-container">
      <div className="search-icon">
        <i className="fa fa-search" />
      </div>

      <input
        type="text"
        placeholder="Search for a country"
        className="search-input"
        value={searchTerm}
        onChange={handleInputValue}
      />
    </section>
  );
};

export default Search;
