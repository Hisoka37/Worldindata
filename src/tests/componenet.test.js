import { render, cleanup } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../Redux/store';
import Filter from '../components/input/filter/Filter';
import Search from '../components/input/search/Search';
import Country from '../components/country/Country';

afterEach(cleanup);
describe('tests for components', () => {
  test('Snapshot test for filter', () => {
    const countryfilter = render(<Provider store={store}><Filter /></Provider>);
    expect(countryfilter).toMatchSnapshot();
  });
  test('Snapshot test for Country', () => {
    const countryDipslayed = render(<Provider store={store}><Country /></Provider>);
    expect(countryDipslayed).toMatchSnapshot();
  });
  test('Snapshot test for Search', () => {
    const countrySearched = render(<Provider store={store}><Search /></Provider>);
    expect(countrySearched).toMatchSnapshot();
  });
});
