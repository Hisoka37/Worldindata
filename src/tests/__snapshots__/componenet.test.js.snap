// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for components Snapshot test for Country 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="country-container"
      >
        <h1>
          Loading .....
        </h1>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="country-container"
    >
      <h1>
        Loading .....
      </h1>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`tests for components Snapshot test for Search 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="search-container"
      >
        <div
          class="search-icon"
        >
          <i
            class="fa fa-search"
          />
        </div>
        <input
          class="search-input"
          placeholder="Search for a country"
          type="text"
          value=""
        />
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="search-container"
    >
      <div
        class="search-icon"
      >
        <i
          class="fa fa-search"
        />
      </div>
      <input
        class="search-input"
        placeholder="Search for a country"
        type="text"
        value=""
      />
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`tests for components Snapshot test for filter 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="filter-container"
      >
        <div
          class="filter"
        >
          <input
            class="filter-input"
            placeholder="Filter by Region"
            readonly=""
            type="text"
            value=""
          />
          <i
            class="fa fa-arrow-down"
          />
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="filter-container"
    >
      <div
        class="filter"
      >
        <input
          class="filter-input"
          placeholder="Filter by Region"
          readonly=""
          type="text"
          value=""
        />
        <i
          class="fa fa-arrow-down"
        />
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
